examples/raspberrypi/rp2xxx/src/rp2040_only/spi_slave.zig

```diff
@@ -0,0 +1,76 @@
+const std = @import("std");
+const microzig = @import("microzig");
+
+const rp2xxx = microzig.hal;
+const time = rp2xxx.time;
+const gpio = rp2xxx.gpio;
+const chip = rp2xxx.compatibility.chip;
+
+const BUF_LEN = 0x100;
+const spi = rp2xxx.spi.instance.SPI0;
+
+const uart = rp2xxx.uart.instance.num(0);
+const baud_rate = 115200;
+const uart_tx_pin = gpio.num(0);
+
+// These will change depending on which GPIO pins you have your SPI device routed to.
+const CS_PIN = 17;
+const SCK_PIN = 18;
+// NOTE: rp2xxx doesn't label pins as MOSI/MISO. Instead a pin is always for
+// either receiving or transmitting SPI data, no matter whether the chip is in
+// master or slave mode.
+const RX_PIN = 16;
+
+pub const microzig_options = .{
+    .log_level = .debug,
+    .logFn = rp2xxx.uart.logFn,
+};
+
+pub fn main() !void {
+    // Set pin functions for CS, SCK, RX
+    const csn = gpio.num(CS_PIN);
+    const mosi = gpio.num(RX_PIN);
+    const sck = gpio.num(SCK_PIN);
+    inline for (&.{ csn, mosi, sck }) |pin| {
+        pin.set_function(.spi);
+    }
+
+    inline for (&.{uart_tx_pin}) |pin| {
+        switch (chip) {
+            .RP2040 => pin.set_function(.uart),
+            .RP2350 => pin.set_function(.uart_first),
+        }
+    }
+    uart.apply(.{
+        .baud_rate = baud_rate,
+        .clock_config = rp2xxx.clock_config,
+    });
+
+    rp2xxx.uart.init_logger(uart);
+
+    std.log.info("Setting SPI as slave device", .{});
+    spi.set_slave(true);
+
+    // Back to 8 bit mode
+    spi.apply(.{
+        .clock_config = rp2xxx.clock_config,
+        .data_width = .eight,
+    }) catch {};
+    var in_buf_eight: [BUF_LEN]u8 = undefined;
+    std.log.info("Reading", .{});
+    spi.read_blocking(u8, 0, &in_buf_eight);
+
+    std.log.info("Got: {s}", .{in_buf_eight});
+
+    // Back to 8 bit mode
+    try spi.apply(.{
+        .clock_config = rp2xxx.clock_config,
+        .data_width = .eight,
+    });
+
+    while (true) {
+        spi.read_blocking(u8, 0, &in_buf_eight);
+        std.log.info("Got: {s}", .{in_buf_eight});
```

Comment:
I changed master to send ascii. If we want it to send bytes with values 00..ff (like the pico-examples) then we should add a hex dump function.